pjsip/src/pjsua-lib/pjsua_core.c

```diff
@@ -667,15 +667,31 @@ static pj_bool_t mod_pjsua_on_rx_request(pjsip_rx_data *rdata)
     pj_bool_t processed = PJ_FALSE;
 
 #if PJSUA_DETECT_MERGED_REQUESTS
-    if (pjsip_tsx_detect_merged_requests(rdata)) {
-        PJ_LOG(4, (THIS_FILE, "Merged request detected"));
+    pjsip_transaction *tsx;
+    if ((tsx = pjsip_tsx_detect_merged_requests(rdata)) != NULL) {
 
-        /* Respond with 482 (Loop Detected) */
-        pjsip_endpt_respond(pjsua_var.endpt, NULL, rdata,
-                            PJSIP_SC_LOOP_DETECTED, NULL,
-                            NULL, NULL, NULL);
+        pjsip_dialog *dlg = pjsip_tsx_get_dlg(tsx);
 
-        return PJ_TRUE;
+        PJ_LOG(4, (THIS_FILE, "Merged request detected (%s) (%s): %s from %s:%d",
+                              dlg ? dlg->obj_name : NULL,
```

Comment:
It is better `""` or `-no-dlg-` rather than `NULL` for the `%s`, which may lead to undefined behavior.